pages/index.tsx

```diff
@@ -114,35 +133,13 @@ export default function Home() {
               }`}
             >
               <Link
-                href={`/register/${domain}`}
+                href={getDomainRoute()}
                 className="flex w-full justify-between  bg-transparent items-center border-gray-200 border-t p-4 pl-5"
               >
                 <p className="text-gray-500 min-h-6">
-                  {domain ? (
-                    domain.includes(".eth") ? (
-                      domain
-                    ) : (
-                      `${domain}.eth`
-                    )
-                  ) : (
-                    <span />
-                  )}
+                  {domain ? domainWithEth(domain) : <span />}
                 </p>
-                {domainError ? (
-                  <Tag colorStyle="redSecondary" size="small">
-                    Invalid name
-                  </Tag>
-                ) : isAvailable === null ? (
-                  <Spinner color="blue" />
-                ) : isAvailable ? (
-                  <Tag colorStyle="greenSecondary" size="small">
-                    Available
-                  </Tag>
-                ) : (
-                  <Tag colorStyle="blueSecondary" size="small">
-                    Registered
-                  </Tag>
-                )}
+                {EnsDomainStatusComponents[domainStatus]}
```

Comment:
This component only carries badges and one icon, can't it be updated to some more explicit name idea?